### ISSUE TITLE
[no-unused-var] Ignore variables that are valid when using ts-check

### ISSUE THREAD

**Issue by muuvmuuv** [typescript-eslint/typescript-eslint#1362]
The issue I have is that `eslint` does not see a type variable that is imported and used with `@ts-check` as "used". I posted an issue on `eslint` already, but they kindly advised me to ask here if that fits more into the typescript rules, which I agree with. In the below example you can see that I enable type checking in a JavaScript file, import a type and use it with the "js"-syntax, unfortunately ESLint can not see that it is used here, which kind of makes sense because it is not really used BUT causes linter errors. Sure, I have not set `typescript-eslint` anywhere in my eslint conf because it is JavaScript (TypeScript is used for the app, JavaScript for scripts only) but was not sure if I need it here too.

> https://github.com/eslint/eslint/issues/12683

**Repro**

```JSON
{
  "root": true,
  "plugins": [
    "prettier"
  ],
  "extends": [
    "eslint:recommended",
    "prettier"
  ],
  "parserOptions": {
    "ecmaVersion": 9,
    "sourceType": "module"
  },
  "env": {
    "node": true,
    "es6": true
  },
  "rules": {
    "prettier/prettier": "error"
  }
}
```

```js
// @ts-check

import { ReleaseType } from 'semver'

/** @type {Array.<ReleaseType>} */
const SEMVER_INCREMENTS = [
  'patch',
  'minor',
  'major',
  'prepatch',
  'preminor',
  'premajor',
  'prerelease',
]
```

**Expected Result**

Should see that ReleaseType is used

**Actual Result**

```
  11:27  error  'ReleaseType' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

`eslint --config scripts/.eslintrc scripts/version/index.js`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.12.0` |
| `@typescript-eslint/parser`        | `2.12.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.7.2` |
| `node`                             | `v12.13.1` |
| `npm`                              | `6.13.3` |

**Comment by bradzacher** [comment]
Try our `no-unused-vars-experimental` rule. This rule is based off of the TS compiler's implementation, so it should handle this out of the box.

----

I don't think this is something we will do within our `@typescript-eslint/no-unused-vars` extension rule.

***Why?***

This would involve parsing JSDoc comments, which is a difficult thing to do correctly, and not something that is supported at all by the parser. It's also something we haven't done at all in any rule, so there would be a large amount of work to make this work.

Additionally, we're on the road to deprecating that rule in favour of the aforementioned experimental version.

All that together with this being based on checks within a JS file (which is a reasonably rare thing within our user base).
